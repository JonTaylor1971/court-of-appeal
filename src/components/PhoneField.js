import React, { Component } from 'react';

class PhoneField extends Component {

    constructor(props) {
        super(props);
        this.handleFieldChange= this.handleFieldChange.bind(this);
        this.validate = this.validate.bind(this);
        this.isValidPhoneNumber = this.isValidPhoneNumber.bind(this);
        this.state = {
            className: "form-field",
            isValid: false,
            title: ''
        };
    }

    render() {
        let autofocus = this.props.autofocus ? true : false;
        return <input
            id={this.props.id}
            name={this.props.name}
            autoFocus={autofocus}
            ref={this.props.ref}
            value={this.props.value || ''}
            onChange={this.handleFieldChange}
            maxLength={17}
            size={30}
            placeholder={this.props.placeHolder}
            readOnly={this.props.readOnly}
            onBlur={this.validate}
            className={this.state.className}
            title={this.state.title}
        />;
    }

    componentDidMount() {
        if (this.props.value) {
            this.validate();
        }
    }

    /**
     * Eliminate all but valid Canadian phone characters are used eg +1(250)385-9999
     * @param e
     */
    handleFieldChange(e) {
        e.target.value = e.target.value.replace(/[^+\d-)( ]/g,'');
        this.props.handleFieldChange(e);
    }

    validate() {
        debugger;
        if (!this.props.value || this.isValidPhoneNumber())
        {
            this.setState({className: "form-field", title: ""}, this.props.validate(true));
        } else {
            this.setState({className: "form-field invalid-field", title: "ERROR: Invalid Phone Number"}, this.props.validate(false));
        }
    }

    isValidPhoneNumber() {
        return this.props.value.match(/^(\+1)?[ ]?\(\d{3}\)[ ]?\d{3}[- ]?\d{4}$/) != null ||
        this.props.value.match(/^(\+1)?[- ]?\d{3}[- ]?\d{3}[- ]?\d{4}$/) != null ||
        this.props.value.match(/^(\+1)?\d{3}-\d{3}-\d{4}$/) != null;
    }


}
export default PhoneField;
