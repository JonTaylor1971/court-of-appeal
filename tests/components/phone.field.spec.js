import React from 'react';
import PhoneField from '../../src/components/PhoneField';
import renderer from 'react-test-renderer';

test('should only allow numeric, dash, and parentheses entries', ()=> {

    let value = '';
    let fieldChanged = (e) => { value = e.target.value; };

    const phoneField = renderer.create(
        <PhoneField id="number-field" value={value}
                      onChange={fieldChanged.bind(this)}/>,
    );
    let tree = phoneField.toJSON();
    expect(tree).toMatchSnapshot();

    let e = { target: { value: "42"}};
    tree.props.onChange(e);
    expect(value).toMatch('42');

    e = { target: { value: "bah" } };
    tree.props.onChange(e);
    expect(value).toMatch('');

    e = { target: { value: "(250)8674309" } };
    tree.props.onChange(e);
    expect(value).toMatch('(250)8674309');

    e = { target: { value: "bah(867)--5309)pooh!" } };
    tree.props.onChange(e);
    expect(value).toMatch('(867)--5309)');

});
