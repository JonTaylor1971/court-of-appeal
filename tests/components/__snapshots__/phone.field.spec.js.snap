// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should only allow numeric, dash, and parentheses entries 1`] = `
<div>
  <input
    autoFocus={false}
    className="form-field"
    id="number-field"
    maxLength={17}
    name={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    size={30}
    title=""
    value=""
  />
  <div
    className="error-message"
  >
     
    
  </div>
</div>
`;
